### DISCUSSION TITLE
Security of APIs

### DISCUSSION THREAD

**Discussion by ShokoufehSimab** [mixcore/mix.core#340] [Q&A] [answered]
Hi, 
I have a question. 
For now, our clients can access all APIs without doing login( from URLs ) . How Can I secure my APIs endpoints and protege my APIs from access without login? 
Best Regards

**Comment by nhathoang989** [comment, marked as answer]
Hi @ShokoufehSimab,
Please pull the latest code from develop,
The APIs, which inherit BaseAuthorizedRestApiController, were Authorized with JWT Bearer token.
Or you can add Authorize Attribute for the API that you want to protect 
Ex: [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]

**Comment by MT-Dev06** [comment]
Hi @nhathoang989,

I'm looking for an api that handle the pages, I found this file  : 
[https://github.com/mixcore/mix.core/blob/master/src/Mix.Cms.Api.RestFul/Controllers/v1/Pages/ApiPagePortalController.cs](url)
but it contains only the endpoint that retrieve all pages, are there other endpoints that retrieve/delete/modify a single page ? thanks

**Reply by nhathoang989** [reply]
Hi @MT-Dev06 

We already implemented all of the restful APIs (GET, PUT, POST, PATCH, DELETE) and some other common methods in BaseAuthorizedRestApiController.

Currently, I just want to override the endpoint that retrieves the list pages In ApiPagePortalContrller.
You can override other endpoints if you want to, 

![image](https://user-images.githubusercontent.com/5327699/107965029-977d6d80-6fdc-11eb-93d3-3657fbeaf2d4.png)
